adapters/huaweiads/huaweiads.go

```diff
@@ -0,0 +1,1028 @@
+package huaweiads
+
+import (
+	"bytes"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/mxmCherry/openrtb/v15/native1"
+	nativeRequests "github.com/mxmCherry/openrtb/v15/native1/request"
+	nativeResponse "github.com/mxmCherry/openrtb/v15/native1/response"
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const HuaweiAdxApiVersion string = "3.4"
+const DefaultBidRequestArrayLength int = 1
+const DefaultCountryName = "ZA"
+const DefaultUnknownNetworkType = 0
+const TimeFormat = "2006-01-02 15:04:05.000"
+const DefaultModelName = "HUAWEI"
+
+// creative type
+const (
+	text        int32 = 1
+	bigPicture  int32 = 2
+	bigPicture2 int32 = 3
+	gif         int32 = 4
+	// used for magazine lock screen
+	noSpecificCreativeType int32 = 5
+	videoText              int32 = 6
+	smallPicture           int32 = 7
+	threeSmallPicturesText int32 = 8
+	video                  int32 = 9
+	iconText               int32 = 10
+	// RewardedVideo only
+	videoWithPicturesText int32 = 11
+	windowAdvertisement   int32 = 13
+)
+
+// ads events
+const (
+	click         string = "click"
+	imp           string = "imp"
+	userClose     string = "userclose"
+	download      string = "download"
+	install       string = "install"
+	downloadstart string = "downloadstart"
+	playStart     string = "playStart"
+	playEnd       string = "playEnd"
+	playResume    string = "playResume"
+	playPause     string = "playPause"
+	appOpen       string = "appOpen"
+)
+
+// ads type
+const (
+	banner        int32 = 8
+	native        int32 = 3
+	rewardedVideo int32 = 7
+	splash        int32 = 1
+	interstitial  int32 = 12
+	roll          int32 = 60
+)
+
+type HuaweiAdsAdapter struct {
```

Comment:
This can just be `type adapter struct {`. The package identifies it as `huaweiads` adapter, and all adapters are referenced as interfaces. The `Builder()` function gives access to an instance of your adapter specifically.

Reply:
Corrected thanks